logprep/processor/calculator/fourFn.py

```diff
@@ -0,0 +1,163 @@
+# pylint: disable=missing-docstring
+# pylint: disable=invalid-name
+"""
+based on https://github.com/pyparsing/pyparsing/blob/master/examples/fourFn.py
+
+Demonstration of the pyparsing module, implementing a simple 4-function expression parser,
+with support for scientific notation, and symbols for e and pi.
+Extended to add exponentiation and simple built-in functions.
+Extended test cases, simplified pushFirst method.
+Removed unnecessary expr.suppress() call (thanks Nathaniel Peterson!), and added Group
+Changed fnumber to use a Regex, which is now the preferred method
+Reformatted to latest pypyparsing features, support multiple and variable args to functions
+Copyright 2003-2019 by Paul McGuire
+"""
+
+import math
+import operator
+
+from pyparsing import (
+    CaselessKeyword,
+    Forward,
+    Group,
+    Literal,
+    Regex,
+    Suppress,
+    Word,
+    alphanums,
+    alphas,
+    delimitedList,
+)
+
+exprStack = []
```

Comment:
In another comment I have already suggested to rewrite this module to a closed class. Another reason is that I am not sure if this is safe for multiprocessing. What happens if two pipelines try to access this imported `exprStack` at the same time?

Reply:
done